src/App.tsx

```diff
@@ -1,11 +1,29 @@
-import { useState } from 'react';
+import { useState, useCallback } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [query, setQuery] = useState('');
+
+  const addMovie = (movie: Movie) => {
+    const index = movies.findIndex(v => v.imdbId === movie.imdbId);
+
+    if (index !== -1) {
+      return;
+    }
+
+    setMovies([
+      ...movies,
+      movie,
+    ]);
```

Comment:
1. move method in if body
2. use callback if you want to work with current state
```suggestion
    if (!isMovieInList) {
       setMovies((prevMovies) => [...prevMovies, movie]);
    }
```